libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/routeprogress/RouteProgressState.kt

```diff
@@ -37,5 +37,7 @@ enum class RouteProgressState {
      * A lack of [android.location.Location] updates from the phone has caused lack of confidence in the
      * progress updates being sent.
      */
-    LOCATION_STALE
+    LOCATION_STALE,
+
+    ROUTE_UNCERTAIN
```

Comment:
Probably it's worth to write documentation in this PR not to make it in a hurry during 1.0

Reply:
This is legacy code, possibly added only to fix compilation errors or make the test app implementation work.
/cc @Guardiola31337

Reply:
Yeah, that's correct. In any case, we can add something around these lines

> State when we start following a route. After a certain number of tracking points we gain confidence and switch to tracking state. We do map-matching rather than route line snapping during this state

cc @mskurydin